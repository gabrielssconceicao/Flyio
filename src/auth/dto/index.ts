import { LoginDto } from './login.dto';
import { LoginResponseDto } from './login-response.dto';
import { RefreshTokenDto } from './refresh-token.dto';
import { TokenPayloadDto } from './token-payload.dto';

export { LoginDto, LoginResponseDto, RefreshTokenDto, TokenPayloadDto };
