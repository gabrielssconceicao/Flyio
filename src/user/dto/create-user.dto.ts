import { ApiProperty } from '@nestjs/swagger';
import {
  IsEmail,
  IsNotEmpty,
  IsOptional,
  IsString,
  MaxLength,
  MinLength,
} from 'class-validator';
export class CreateUserDto {
  @ApiProperty({
    example: 'John Doe',
    description: 'User name',
    minLength: 3,
    maxLength: 255,
  })
  @IsString()
  @IsNotEmpty()
  @MinLength(3)
  @MaxLength(255)
  name: string;

  @ApiProperty({
    example: 'jdoe@email.com',
    description: 'User email',
  })
  @IsEmail()
  email: string;

  @ApiProperty({
    example: '123456',
    description: 'User password',
    minLength: 6,
    maxLength: 255,
  })
  @IsString()
  @IsNotEmpty()
  @MinLength(6)
  @MaxLength(255)
  password: string;

  @ApiProperty({
    example: 'This is my bio',
    description: 'User bio',
    maxLength: 255,
  })
  @IsOptional()
  @IsString()
  @MaxLength(255)
  bio?: string;

  @ApiProperty({
    example: 'example.png',
    description: 'User profile image',
  })
  @IsOptional()
  @IsString()
  profileImg?: string;
}
