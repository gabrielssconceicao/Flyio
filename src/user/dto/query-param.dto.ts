import { Type } from 'class-transformer';
import { IsOptional, IsString, Max, Min } from 'class-validator';

export class QueryParamDto {
  @IsOptional()
  @IsString()
  search?: string;
  @IsOptional()
  @Min(1)
  @Max(50)
  @Type(() => Number)
  limit?: number;
  @IsOptional()
  @Min(0)
  @Type(() => Number)
  offset?: number;
}
