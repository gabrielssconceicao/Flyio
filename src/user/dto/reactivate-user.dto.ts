import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class ReactivateUserDto {
  @ApiProperty({
    example: 'krcio48hgGYIFYJfuyfJYAYu',
    description: 'Activation token',
  })
  @IsNotEmpty({ message: dtoErrorMessages('Token', DtoErrorType.REQUIRED) })
  @IsString()
  token: string;
}
