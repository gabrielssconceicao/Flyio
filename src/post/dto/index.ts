import { CreatePostDto } from './create-post.dto';
import { FindAllPostsResponseDto } from './find-all-posts.dto';
import { FindAllLikedPostsResponseDto } from './find-all-liked-post.dto';
export { CreatePostDto, FindAllPostsResponseDto, FindAllLikedPostsResponseDto };
