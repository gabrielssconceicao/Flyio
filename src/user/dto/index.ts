import { CreateUserDto } from './create-user.dto';
import { FindAllUsersResponseDto } from './find-all-users.dto';
import { QueryParamDto } from './query-param.dto';
import { ReactivateUserDto } from './reactivate-user.dto';
import { UpdateUserDto } from './update-user.dto';

export {
  CreateUserDto,
  FindAllUsersResponseDto,
  QueryParamDto,
  ReactivateUserDto,
  UpdateUserDto,
};
