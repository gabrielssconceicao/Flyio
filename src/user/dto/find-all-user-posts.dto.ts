import { ApiProperty, OmitType } from '@nestjs/swagger';
import { FindAllResponseDto } from 'src/common/dto/find-all-response.dto';
import { PostEntity } from '../../post/entities/post.entity';

export class FindAllUsersPostsResponseDto extends FindAllResponseDto<
  Omit<PostEntity, 'user'>
> {
  @ApiProperty({
    type: [OmitType(PostEntity, ['user'])],
    description: 'List of posts of a user with limited details',
  })
  items: Omit<PostEntity, 'user'>[];
}
