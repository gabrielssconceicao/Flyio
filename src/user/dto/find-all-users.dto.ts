import { ApiProperty } from '@nestjs/swagger';
import { PickType } from '@nestjs/mapped-types';
import { User } from '../entities/user.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

class UserSummaryDto extends PickType(User, [
  'id',
  'name',
  'username',
  'profileImg',
] as const) {
  @ApiProperty({
    example: 'd-43-5df-df4',
    description: 'Unique identifier of the user',
  })
  id: string;

  @ApiProperty({ example: 'John Doe', description: 'Name of the user' })
  name: string;

  @ApiProperty({ example: 'jdoe', description: 'Username of the user' })
  username: string;

  @ApiProperty({
    example: 'https://example.com/profile.jpg',
    description: 'Profile image URL',
    nullable: true,
  })
  profileImg: string | null;
}

export class FindAllUsersResponseDto extends FindAllResponseDto<UserSummaryDto> {
  @ApiProperty({
    type: [UserSummaryDto],
    description: 'List of users with limited details',
  })
  items: UserSummaryDto[];
}
