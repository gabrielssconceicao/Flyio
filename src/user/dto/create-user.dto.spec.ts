<<<<<<< HEAD
import { validate } from 'class-validator';
import { CreateUserDto } from './create-user.dto';
import userMock from 'src/mocks/user.mock';

const fillDto = (dto: CreateUserDto) => {
  dto.name = userMock.name;
  dto.email = userMock.email;
  dto.password = userMock.password;
  return dto;
};

describe('<CreateUserDto>', () => {
  it('should create a valid DTO without bio and profileImg', async () => {
    const dto = new CreateUserDto();
    fillDto(dto);

=======
import { validate, ValidationError } from 'class-validator';
import { CreateUserDto } from './create-user.dto';
function formatErrors(errors: ValidationError[]) {
  return errors.map((error) => ({
    property: error.property,
    constraints: error.constraints,
    value: error.value,
  }));
}
describe('<CreateUserDto>', () => {
  let dto: CreateUserDto;

  beforeEach(() => {
    dto = new CreateUserDto();

    dto.name = 'John Doe';
    dto.username = 'jDoe453';
    dto.email = 'jdoe@me.com';
    dto.password = 'exampl3P@ssw0rd';
    dto.bio = 'This is my bio';
  });

  it('should validate a DTO', async () => {
>>>>>>> images
    const errors = await validate(dto);
    expect(errors.length).toBe(0);
  });

<<<<<<< HEAD
  describe('<Property: Name />', () => {
    it('should fail if name is empty', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.name = '';

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('name');
    });
    it('should fail if name is too short', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.name = 'Jo';

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('name');
    });
    it('should fail if name is too long', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.name = 'Bob'.repeat(100);

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('name');
    });
  });

  describe('<Property: Email />', () => {
    it('should fail if email is invalid', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.email = 'invalid-email';
=======
  describe('<Property: Name>', () => {
    it('should fail if name is empty', async () => {
      dto.name = '';
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('name');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
    it('should fail if name is too short', async () => {
      dto.name = 'Jo';
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('name');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
    it('should fail if name is too long', async () => {
      dto.name = 'Jonh'.repeat(100);
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('name');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
  });

  describe('<Property: Username>', () => {
    it('should fail if username is empty', async () => {
      dto.username = null;
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('username');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
    it('should fail if username is too short', async () => {
      dto.username = 'jD';
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('username');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
    it('should fail if username is too long', async () => {
      dto.username = 'jDoe42'.repeat(100);
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('username');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
  });

  describe('<Property: Email>', () => {
    it('should fail if email is empty', async () => {
      dto.email = null;
>>>>>>> images

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('email');
<<<<<<< HEAD
    });
  });

  describe('<Property: Password />', () => {
    it('should fail if password is empty', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.password = '';

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('password');
    });
    it('should fail if password is too short', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.password = '1234';

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('password');
    });
    it('should fail if password is too long', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.password = '1'.repeat(256);

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('password');
    });
  });

  describe('<Property: Bio />', () => {
    it('should create a DTO with bio', async () => {
      const createDto = new CreateUserDto();
      fillDto(createDto);
      createDto.bio = userMock.bio;
      const errors = await validate(createDto);
      expect(errors.length).toBe(0);
    });
    it('should fail if bio is too long', async () => {
      const dto = new CreateUserDto();
      fillDto(dto);
      dto.bio = 'This is my bio'.repeat(100);

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('bio');
    });
  });
  describe('<Property: ProfileImg />', () => {
    it('should create a DTO with profileImg', async () => {
      const createDto = new CreateUserDto();
      fillDto(createDto);
      createDto.profileImg = userMock.profileImg;
      const errors = await validate(createDto);
      expect(errors.length).toBe(0);
=======
      expect(formatErrors(errors)).toMatchSnapshot();
    });
    it('should fail if email is not valid', async () => {
      dto.email = 'jdoe@me';

      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('email');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
  });

  describe('<Property: Password>', () => {
    it('should fail if password is empty', async () => {
      dto.password = null;
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('password');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
    it('should fail if password is too short', async () => {
      dto.password = '12345';
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('password');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
    it('should fail if password is too long', async () => {
      dto.password = '1'.repeat(101);
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('password');
      expect(formatErrors(errors)).toMatchSnapshot();
    });
  });

  describe('<Property: Bio>', () => {
    it('should validate a DTO if bio is not send', async () => {
      dto.bio = undefined;
      const errors = await validate(dto);
      expect(errors.length).toBe(0);
    });
    it('should validate a DTO if bio too long', async () => {
      dto.bio = 'b'.repeat(256);
      const errors = await validate(dto);
      expect(errors.length).toBe(1);
      expect(errors[0].property).toBe('bio');
      expect(formatErrors(errors)).toMatchSnapshot();
>>>>>>> images
    });
  });
});
