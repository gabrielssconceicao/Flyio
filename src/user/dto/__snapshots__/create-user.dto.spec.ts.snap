// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateUserDto> <Property: Bio> should validate a DTO if bio too long 1`] = `
[
  {
    "constraints": {
      "maxLength": "Bio must be at most 255 characters long",
    },
    "property": "bio",
    "value": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
  },
]
`;

exports[`<CreateUserDto> <Property: Email> should fail if email is empty 1`] = `
[
  {
    "constraints": {
      "isEmail": "email must be an email",
      "isNotEmpty": "Email is required",
    },
    "property": "email",
    "value": null,
  },
]
`;

exports[`<CreateUserDto> <Property: Email> should fail if email is not valid 1`] = `
[
  {
    "constraints": {
      "isEmail": "email must be an email",
    },
    "property": "email",
    "value": "jdoe@me",
  },
]
`;

exports[`<CreateUserDto> <Property: Name> should fail if name is empty 1`] = `
[
  {
    "constraints": {
      "isNotEmpty": "Name is required",
      "minLength": "Name must be at least 3 characters long",
    },
    "property": "name",
    "value": "",
  },
]
`;

exports[`<CreateUserDto> <Property: Name> should fail if name is too long 1`] = `
[
  {
    "constraints": {
      "maxLength": "Name must be at most 100 characters long",
    },
    "property": "name",
    "value": "JonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonhJonh",
  },
]
`;

exports[`<CreateUserDto> <Property: Name> should fail if name is too short 1`] = `
[
  {
    "constraints": {
      "minLength": "Name must be at least 3 characters long",
    },
    "property": "name",
    "value": "Jo",
  },
]
`;

exports[`<CreateUserDto> <Property: Password> should fail if password is empty 1`] = `
[
  {
    "constraints": {
      "isNotEmpty": "Password is required",
      "isString": "Password must be a string",
      "maxLength": "Password must be at most 100 characters long",
      "minLength": "Password must be at least 6 characters long",
    },
    "property": "password",
    "value": null,
  },
]
`;

exports[`<CreateUserDto> <Property: Password> should fail if password is too long 1`] = `
[
  {
    "constraints": {
      "maxLength": "Password must be at most 100 characters long",
    },
    "property": "password",
    "value": "11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111",
  },
]
`;

exports[`<CreateUserDto> <Property: Password> should fail if password is too short 1`] = `
[
  {
    "constraints": {
      "minLength": "Password must be at least 6 characters long",
    },
    "property": "password",
    "value": "12345",
  },
]
`;

exports[`<CreateUserDto> <Property: Username> should fail if username is empty 1`] = `
[
  {
    "constraints": {
      "isNotEmpty": "Username is required",
      "isString": "Username must be a string",
      "maxLength": "Username must be at most 100 characters long",
      "minLength": "Username must be at least 3 characters long",
    },
    "property": "username",
    "value": null,
  },
]
`;

exports[`<CreateUserDto> <Property: Username> should fail if username is too long 1`] = `
[
  {
    "constraints": {
      "maxLength": "Username must be at most 100 characters long",
    },
    "property": "username",
    "value": "jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42jDoe42",
  },
]
`;

exports[`<CreateUserDto> <Property: Username> should fail if username is too short 1`] = `
[
  {
    "constraints": {
      "minLength": "Username must be at least 3 characters long",
    },
    "property": "username",
    "value": "jD",
  },
]
`;
